src/Xamarin.Android.Build.Tasks/Tasks/FindLayoutsToBind.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks
+{
+	public class FindLayoutsToBind : Task
+	{
+		static readonly Regex layoutPathRegex = new Regex ($".*{Path.DirectorySeparatorChar}layout(-?\\w+)*{Path.DirectorySeparatorChar}.*\\.a?xml$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Comment:
Use of `Path.DirectorySeparatorChar` within a Regex is *DOOMED*, because Windows.

Also, if you're using a Regex, prefer `@"..."` strings so you don't need to escape `\`. :-)

Perhaps you could "shadow" `DirectorySeparatorChar`?

```csharp
static string RegexDirectorySeparator {
  get {
    if (Path.DirectorySeparatorChar == '\\')
      return @"\\";
    return "/";
  }
}
```

*Additionally*, it's not unusual in my experience for directory separator chars to be "duplicated", e.g. "foo//bar" instead of "foo/bar". This Regex won't properly cope with that, so it should *probably* use `$"{RegexDirectorySeparator}+"` to permit repeats.